include/heuristic/HeuristicMapper.hpp

```diff
@@ -151,6 +193,99 @@ class HeuristicMapper : public Mapper {
       }
 
       swaps.back().emplace_back(source, target, middleAnc, qc::Teleportation);
+
+      if (considerFidelity) {
+        throw QMAPException(
+            "Teleportation currently not supported for noise-aware mapping");
+        // TODO: check this cost implementation and add teleportation support in
+        // noise-aware heuristic accounting for fidelity difference of single
+        // qubit gates (two qubit gates are handled in the heuristic)
+        /*costFixed += (-(singleQubitGateMultiplicity.at(q2) -
+        singleQubitGateMultiplicity.at(q1)) *
+                         log2(singleQubitFidelities.at(swap.first))
+                      -(singleQubitGateMultiplicity.at(q1) -
+        singleQubitGateMultiplicity.at(q2)) *
+                        log2(singleQubitFidelities.at(swap.second))
+                      );
+        // adding cost oft the swap gate
+        costFixed += (-    log2(twoQubitFidelities.at(source).at(middleAnc))
+                      -    log2(twoQubitFidelities.at(middleAnc).at(target))
+                      -    log2(singleQubitFidelities.at(source))
+                      -    log2(singleQubitFidelities.at(middleAnc))
+                      -2 * log2(singleQubitFidelities.at(target))
+                      -    log2(1.0 -
+        arch.getProperties().readoutErrorRate.get(source))
+                      -    log2(1.0 -
+        arch.getProperties().readoutErrorRate.get(middleAnc)));*/
+      } else {
+        costFixed += COST_TELEPORTATION;
+      }
+    }
+
+    /**
+     * @brief recalculates the fixed cost of the node from current mapping and
+     * swaps
+     *
+     * @param arch the architecture for calculating distances between physical
+     * qubits and supplying qubit information such as fidelity
+     * @param singleQubitGateMultiplicity vector containing the number of gates
+     * acting on each logical qubit
+     * @param considerFidelity controls if qubit fidelity should be taken into
+     * account
+     */
+    void recalculateFixedCost(
+        const Architecture&              arch,
+        const std::vector<std::uint16_t> singleQubitGateMultiplicity,
+        bool                             considerFidelity) {
+      const auto& singleQubitFidelities = arch.getSingleQubitFidelities();
+      const auto& twoQubitFidelities    = arch.getFidelityTable();
+
+      costFixed = 0;
+      if (considerFidelity) {
+        // adding costs of single qubit gates
+        for (std::uint16_t i = 0U; i < arch.getNqubits(); ++i) {
+          if (singleQubitGateMultiplicity.at(i) == 0) {
+            continue;
+          }
+          costFixed += -singleQubitGateMultiplicity.at(i) *
```

Comment:
This was a (poor) attempt of increasing readability. As I am using the -log2 as fidelity-aware cost everywhere, and I thought it would be more obvious to be just a negating sign of the log resulting overall in an addition of a positive cost.
However, just a comment would probably be more helpful.